gapis/api/vulkan/graph_visualization.go

```diff
@@ -88,51 +92,52 @@ func getMaxCommonPrefix(label1 *api.Label, label2 *api.Label) int {
 	return size
 }
 
-func addDebugMarker(builder *labelForVulkanCommands, from, to int) {
-	level := builder.labelsInsideDebugMarkers[len(builder.labelsInsideDebugMarkers)-1].GetSize() - 1
-	builder.numberOfDebugMarker++
+func (dm *debugMarker) addDebugMarker(from, to int) {
```

Comment:
I still don't really understand how addDebugMarker and checkDebugMarkers are working. A few comments might help.

Reply:
I implemented a different way to handle the debug markers, I think now it is very simple to understand the idea, also I added comments to better understanding.